### PR TITLE
 Scale trait and move to u32 blocknumbers

### PR DESCRIPTION
**Pull request by gavofyork**

### CODE REVIEW

File: node/runtime/src/lib.rs [hunk 1]
```diff
@@ -179,7 +179,7 @@ impl timestamp::Trait for Runtime {
 }
 
 parameter_types! {
-	pub const UncleGenerations: u64 = 5;
+	pub const UncleGenerations: u32 = 5;
```
Comment (bkchr) [inline node/runtime/src/lib.rs]:
```suggestion
	pub const UncleGenerations: BlockNumber = 5;
```

**Comment by andresilva** [comment]
> Looks good.
> 
> Why do we do this?

With 6 seconds block time `u32` should be good for about 800 years.

File: core/sr-primitives/src/traits.rs [hunk 2]
```diff
@@ -303,6 +303,47 @@ pub trait CheckedConversion {
 }
 impl<T: Sized> CheckedConversion for T {}
 
+/// Multiply and devide by a number that isn't necessarily the same type. Basically just the same
```
Comment (shawntabrizi) [inline core/sr-primitives/src/traits.rs]:
```suggestion
/// Multiply and divide by a number that isn't necessarily the same type. Basically just the same
```

Comment (shawntabrizi) [reply core/sr-primitives/src/traits.rs]: It feels a little unfortunate we have a SCALE codec and a Scale trait, but that may just be me.

**Pull request merged**